diff --git a/dist/classes/Agent.js b/dist/classes/Agent.js
index 4a31fe58d6b2226252e8f130486f7a865b7c7aa8..b0d819804c867f268d1b412984315a3ac6d0928d 100644
--- a/dist/classes/Agent.js
+++ b/dist/classes/Agent.js
@@ -153,7 +153,7 @@ class Agent {
         // > are also accepted:
         // >   ca, cert, ciphers, clientCertEngine, crl, dhparam, ecdhCurve, honorCipherOrder,
         // >   key, passphrase, pfx, rejectUnauthorized, secureOptions, secureProtocol, servername, sessionIdContext.
-        if (configuration.secureEndpoint) {
+        if (this.protocol === 'https:') {
             // Determine servername - Node.js doesn't allow IP addresses as servername
             const host = (_f = configuration.servername) !== null && _f !== void 0 ? _f : connectionConfiguration.host;
             const servername = net_1.default.isIP(host) ? undefined : host;
